Assets/MixedRealityToolkit.Tests/PlayModeTests/NearInteractionTouchableTests.cs

```diff
@@ -0,0 +1,245 @@
+    {
+        /// <summary>
+        /// Test creating adding a NearInteractionTouchable to GameObject programmatically.
+        /// Should be able to run scene without getting any exceptions.
+        /// </summary>
+        /// <returns></returns>
+        [UnityTest]
+        public IEnumerator Test01_ManipulationHandlerInstantiate()
+        {
+            var testObject = GameObject.CreatePrimitive(PrimitiveType.Cube);
+            testObject.transform.localScale = Vector3.one * 0.2f;
+
+            var touchable = testObject.AddComponent<NearInteractionTouchable>();
+            // Wait for two frames to make sure we don't get null pointer exception.
+            yield return null;
+            yield return null;
+
+            GameObject.Destroy(testObject);
+            // Wait for a frame to give Unity a change to actually destroy the object
+            yield return null;
+        }
+
+        private InputSimulationService inputSim;
+        IMixedRealityInputSystem inputSystem;
+
+        [SetUp]
+        public void SetupMrtk()
+        {
+            TestUtilities.InitializeMixedRealityToolkitAndCreateScenes(true);
+            TestUtilities.InitializePlayspace();
+
+            inputSim = PlayModeTestUtilities.GetInputSimulationService();
+            Assert.NotNull(inputSim);
+
+            Assert.IsTrue(MixedRealityServiceRegistry.TryGetService<IMixedRealityInputSystem>(out inputSystem));
+            Assert.NotNull(inputSystem);
+        }
+
+        [TearDown]
+        public void ShutdownMrtk()
+        {
+            TestUtilities.ShutdownMixedRealityToolkit();
+        }
+
+        private const int numSteps = 30;
+
+        // Scale larger than bounds vector to test bounds checks
+        private float objectScale = 0.4f;
+        private Vector3 initialHandPosition = new Vector3(0, 0, 0.5f);
+        private Vector3 objectPosition = new Vector3(0f, 0f, 1f);
+        private Vector3 rightPosition = new Vector3(1f, 0f, 1f);
+        private Vector3 backPosition = new Vector3(0f, 0f, 2f);
+        private Vector3 outOfBoundsOffset = new Vector3(0.15f, 0f, 0f);
+        private Vector3 touchNormal = new Vector3(0, 0, -1);
+
+        private T CreateTouchable<T>() where T : BaseNearInteractionTouchable
+        {
+            // Set up cube with touchable
+            var testObject = GameObject.CreatePrimitive(PrimitiveType.Cube);
+            testObject.transform.localScale = Vector3.one * objectScale;
+            testObject.transform.position = objectPosition;
+
+            var touchable = testObject.AddComponent<T>();
+
+            // Register the game object to receive events
+            inputSystem.Register(testObject);
+
+            return touchable;
+        }
+
+        /// <summary>
+        /// Test creates an object with NearInteractionTouchable
+        /// </summary>
+        /// <returns></returns>
+        [UnityTest]
+        public IEnumerator Test02_NearInteractionTouchable()
```

Comment:
@wiwei to keep me honest but I believe our policy is to not name tests with Test0X_blah and just use 'blah'. We should probably write that down somewhere.

Reply:
Fixed